packages/fuselage-ui-kit/src/elements/MarkdownTextElement.tsx

```diff
@@ -0,0 +1,24 @@
+import { useTranslation } from '@rocket.chat/ui-contexts';
+import { parse } from '@rocket.chat/message-parser';
+import { Markup } from '@rocket.chat/gazzodown';
+import type { TextObject } from '@rocket.chat/ui-kit';
+
+import { useUiKitContext } from '../contexts/kitContext';
+
+const MarkdownTextElement = ({ textObject }: { textObject: TextObject }) => {
+  const t = useTranslation() as (
+    key: string,
+    args: { [key: string]: string | number }
+  ) => string;
+  const { appId } = useUiKitContext();
+
+  const { i18n } = textObject;
+
+  if (i18n) {
+    textObject.text = t(`apps-${appId}-${i18n.key}`, { ...i18n.args });
```

Comment:
It's not safe to mutate an object coming from a prop. Sometimes the `props` object is frozen and shared. Consider it read-only.